### PR TITLE
Expose PulpRemoteUserAuthentication to plugin writers

### LINKED ISSUE #2262: Expose PulpRemoteUserAuthentication to plugin writers
This will allow plugin writers to take a rest_framework remote authentication class into account when building protected endpoints.

### PR DESCRIPTION
**Pull request by lubosmj**
closes #2262

### CODE REVIEW

**Comment by lubosmj** [comment]
The change is required for pulp_container where we need to consider users who were authenticated remotely by a web server since the access to specific endpoints is managed exceptionally in the plugin.

File: CHANGES/plugin_api/2262.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Exposed the ``PulpRemoteUserAuthentication`` class to plugin writers. This will enable them to
+consider remote authentication scenarios when building protected endpoints.
```
Comment (ipanova) [inline CHANGES/plugin_api/2262.feature:2]: you can shorten it with: "This enables remote authentication scenarios.."

Comment (lubosmj) [reply CHANGES/plugin_api/2262.feature:2]: This is not correct. It does not enable authentication scenarios itself.

You as a plugin writer will have rather a chance to override or utilize such a class if you want to manage access for particular endpoints (like we do in pulp_container).

Comment (ipanova) [reply CHANGES/plugin_api/2262.feature:2]: `to enable` in this context is read as 'to make possible remote authentication scenarios'

Comment (daviddavis) [reply CHANGES/plugin_api/2262.feature:2]: "This will allow them to use remote authentication methods when building protected endpoints"

maybe?

**Review by ipanova — APPROVED** [review]

File: pulpcore/plugin/authentication/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from pulpcore.app.authentication import PulpRemoteUserAuthentication  # noqa
```
Comment (ipanova) [inline pulpcore/plugin/authentication/__init__.py:1]: i think it should follow this format https://github.com/pulp/pulpcore/blob/34b490c50594bb62bdc99951acd96734beb5f5e2/pulpcore/plugin/access_policy.py

**Review by mdellweg — APPROVED** [review]

**Review by fao89 — APPROVED** [review]

**Pull request merged**